engine/pkg/externalresource/broker/broker.go

```diff
@@ -340,23 +341,27 @@ func (b *DefaultBroker) getPersistResource(
 			WorkerID:    record.Worker,   /* creator id*/
 		},
 	}
-	var desc internal.ResourceDescriptor
-
-	if options.cleanBeforeOpen {
-		err := fm.RemoveResource(ctx, ident)
-		// LocalFileManager may return ErrResourceDoesNotExist, which can be
-		// ignored because the resource no longer exists.
-		if err != nil && !derrors.ErrResourceDoesNotExist.Equal(err) {
-			return nil, err
-		}
-		desc, err = fm.CreateResource(ctx, ident)
-		if err != nil {
-			return nil, err
-		}
-		return desc, nil
```

Comment:
What's wrong with the old logic? Remove and Create seems to be the most straightforward approach to clean the resource.

Reply:
The core problem, in terms of interface semantics, is that `CreateResource ` can only be used to create resources that belong to the current executor.
ref: https://github.com/pingcap/tiflow/blob/master/engine/pkg/externalresource/internal/s3/file_manager.go#L95